nixos/modules/services/networking/harmonia.nix

```diff
@@ -0,0 +1,100 @@
+{ config, pkgs, lib, ... }:
+let
+  cfg = config.services.harmonia;
+
+  format = pkgs.formats.toml { };
+  configFile = format.generate "harmonia.toml" cfg.settings;
+in
+{
+  options = {
+    services.harmonia = {
+      enable = lib.mkEnableOption (lib.mdDoc "Harmonia: Nix binary cache written in Rust");
+
+      signKeyPath = lib.mkOption {
+        type = lib.types.nullOr lib.types.path;
+        default = null;
+        description = lib.mdDoc "Path to the signing key to use for signing the cache";
+      };
+
+      settings = lib.mkOption {
+        inherit (format) type;
+
+        description = lib.mdDoc "Settings to merge with the default configuration";
+      };
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+    services.harmonia.settings = builtins.mapAttrs (_: v: lib.mkDefault v) {
+      bind = "[::]:5000";
```

Comment:
Do you need to specify the default explicitly? Will they not be used without setting s config file?

Reply:
Done.